org.idempiere.test/src/org/idempiere/test/model/Allocation2ndAcctSchemaTest.java

```diff
@@ -1057,6 +1059,332 @@ public void testAllocateInvoicePaymentPosting_10() {
+			gainLossAmt = new BigDecimal(100).setScale(eur.getStdPrecision(), RoundingMode.HALF_UP);
+			gainLossLineList.add(new PostingLine(eur, gainLossAmt, Env.ZERO));
+			tradeLineList.add(new PostingLine(eur, Env.ZERO, gainLossAmt));
+			
+			testAllocationPosting(ass, allocList, paymentLineList, tradeLineList, gainLossLineList, null);
+			
+			reverseDocument(invoice, false);
+			MInvoice reversalInvoice = new MInvoice(Env.getCtx(), invoice.getReversal_ID(), getTrxName());
+			postDocument(reversalInvoice);
+			
+			MAllocationHdr reversalAlloc = null;
+			allocList = MAllocationHdr.getOfInvoice(Env.getCtx(), invoice.getC_Invoice_ID(), getTrxName());
+			for (MAllocationHdr allocation : allocList) {
+				if (allocation.getReversal_ID() > 0 && allocation.getReversal_ID() < allocation.get_ID()) {
+					reversalAlloc = allocation;
+					break;
+				}
+			}
+			
+			if (reversalAlloc == null)
+				reversalAlloc = new MAllocationHdr(Env.getCtx(), alloc.getReversal_ID(), getTrxName());
+		
+			allocList = new MAllocationHdr[] { reversalAlloc };
+			
+			paymentLineList = new ArrayList<PostingLine>();
+			tradeLineList = new ArrayList<PostingLine>();
+			gainLossLineList = new ArrayList<PostingLine>();
+			accountedDrAmt = getAccountedAmount(usd, allocAmount, cr1b.getMultiplyRate());
+			accountedCrAmt = getAccountedAmount(usd, allocAmount, cr1b.getMultiplyRate());
+			tradeLineList.add(new PostingLine(usd, accountedDrAmt, Env.ZERO));
+			paymentLineList.add(new PostingLine(usd, Env.ZERO, accountedCrAmt));
+			gainLossAmt = new BigDecimal(100).setScale(usd.getStdPrecision(), RoundingMode.HALF_UP);
+			gainLossLineList.add(new PostingLine(usd, Env.ZERO, gainLossAmt));
+			tradeLineList.add(new PostingLine(usd, gainLossAmt, Env.ZERO));
+			
+			testAllocationPosting(ass, allocList, paymentLineList, tradeLineList, gainLossLineList, null);
+			
+			paymentLineList = new ArrayList<PostingLine>();
+			tradeLineList = new ArrayList<PostingLine>();
+			gainLossLineList = new ArrayList<PostingLine>();
+			accountedDrAmt = getAccountedAmount(eur, allocAmount, cr1a.getMultiplyRate());
+			accountedCrAmt = getAccountedAmount(eur, allocAmount, cr1a.getMultiplyRate());
+			tradeLineList.add(new PostingLine(eur, accountedDrAmt, Env.ZERO));
+			paymentLineList.add(new PostingLine(eur, Env.ZERO, accountedCrAmt));
+			gainLossAmt = new BigDecimal(100).setScale(eur.getStdPrecision(), RoundingMode.HALF_UP);
+			gainLossLineList.add(new PostingLine(eur, Env.ZERO, gainLossAmt));
+			tradeLineList.add(new PostingLine(eur, gainLossAmt, Env.ZERO));
+			
+			testAllocationPosting(ass, allocList, paymentLineList, tradeLineList, gainLossLineList, null);
+		} finally {
+			deleteConversionRate(cr1a);
+			deleteConversionRate(cr1b);
+			deleteConversionRate(cr2a);
+			deleteConversionRate(cr2b);
+			deleteConversionRate(cr3a);
+			deleteConversionRate(cr3b);
+			
+			plv.deleteEx(true);
+			priceList.deleteEx(true);
+			
+			rollback();
```

Comment:
please move rollback() above deleteConversionRate(cr1a), this will give problem with Oracle.